src/components/Basket.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { FC } from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import * as actions from '../redux/actions';
+import { BackPath } from './BackPath';
+
+interface Props {
+  basket: Basket[];
+  setBasket: (basket: Basket[]) => void;
+}
+
+const BasketTemplate: FC<Props> = ({
+  basket,
+  setBasket,
+}) => {
+  const removeItem = (id: string): void => {
+    setBasket([...basket.filter(item => item.id !== id)]);
+  };
+
+  const onIncrement = (id: string): void => {
+    setBasket([...basket.map(item => {
+      if (item.id === id) {
+        return {
+          ...item,
+          quantity: item.quantity + 1,
+        };
+      }
+
+      return item;
+    })]);
+  };
+
+  const onDecrement = (id: string): void => {
+    setBasket([...basket.map(item => {
+      if (item.id === id) {
+        return {
+          ...item,
+          quantity: item.quantity - 1,
+        };
+      }
+
+      return item;
+    })]);
+  };
+
+  if (basket.length > 0) {
+    return (
+      <div className="basket">
+        <BackPath />
+        <h2>Cart</h2>
+
+        <ul>
+          {basket.length && (basket.map(item => (
```

Comment:
в этом случае если массив будет пустой, тебе нарисуется 0 на странице
```suggestion
          {basket.length > 0 && (basket.map(item => (
```